crates/stages/api/src/pipeline/mod.rs

```diff
@@ -63,14 +64,14 @@ pub type PipelineWithResult<DB> = (Pipeline<DB>, Result<ControlFlow, PipelineErr
 /// # Defaults
 ///
 /// The [`DefaultStages`](crate::sets::DefaultStages) are used to fully sync reth.
-pub struct Pipeline<DB: Database> {
+pub struct Pipeline<N: NodeTypesWithDB> {
```

Comment:
not clear to me why we need the full node types here?

Reply:
basically it's needed because pipeline currently depends on concrete `ProviderFactory` which needs chainspec AT 

we can indeed avoid this by doing `<DB, Provider: DatabaseProviderFactory<DB>>`, but given that we're going to make providers generic over primitive types this DB generic would at some point become `NodeTypesWithDB` or equivalent anyway